runtime/compiler/x/codegen/X86PrivateLinkage.cpp

```diff
@@ -2473,8 +2473,10 @@ TR::Register *J9::X86::PrivateLinkage::buildCallPostconditions(TR::X86CallSite &
             }
          }
 
-      TR_LiveRegisters *lr = cg()->getLiveRegisters(TR_FPR);
-      if(!lr || lr->getNumberOfLiveRegisters() > 0)
+      TR_LiveRegisters *liveFPRs = cg()->getLiveRegisters(TR_FPR);
+      TR_LiveRegisters *liveVRFs = cg()->getLiveRegisters(TR_VRF);
+
+      if (!liveFPRs || liveFPRs->getNumberOfLiveRegisters() > 0 || !liveVRFs || liveVRFs->getNumberOfLiveRegisters() > 0)
          {
          for (regIndex = TR::RealRegister::FirstXMMR; regIndex <= TR::RealRegister::LastXMMR; regIndex = (TR::RealRegister::RegNum)(regIndex + 1))
```

Comment:
This will only preserve up to LastXMMR, which seems to be fixed at xmm15 regardless of the capabilities of the target processor.  What about AVX-512 that goes up to zmm31?  Or are you safe because register assignment is currently limited to zmm0-15 ?

https://github.com/eclipse-omr/omr/blob/6c925211b0314455e94090e0e49d3aa3c3e495fc/compiler/x/amd64/codegen/RealRegisterEnum.hpp#L87